--- rootfs.original/home/lava/klipper/klippy/extras/filament_detect.py
+++ rootfs/home/lava/klipper/klippy/extras/filament_detect.py
@@ -52,6 +52,9 @@
         self.printer.register_event_handler("filament_feed:port", self._feed_port_evt_handle)
         self.printer.register_event_handler("filament_switch_sensor:runout", self._runout_evt_handle)
 
+        self.webhooks = self.printer.lookup_object('webhooks')
+        self.webhooks.register_endpoint("filament_detect/set", self._handle_filament_detect_set)
+
     def _ready(self):
         self.filament_feed_objects = self.printer.lookup_objects('filament_feed')
         self._fm175xx_reader = self.printer.lookup_object('fm175xx_reader')
@@ -173,6 +176,32 @@
 
         return error, info
 
+    def _handle_filament_detect_set(self, web_request):
+        try:
+            channel = web_request.get_int('channel', None)
+            if channel is None:
+                raise web_request.error("channel must be specified!")
+
+            if (channel < 0 or channel >= self._channel_nums):
+                raise web_request.error("channel[%d] is out of range[0, %d]"
+                                        % (channel, self._channel_nums - 1))
+
+            params = web_request.get_dict('info')
+
+            filament_info = copy.deepcopy(filament_protocol.FILAMENT_INFO_STRUCT)
+            for key, default_value in filament_protocol.FILAMENT_INFO_STRUCT.items():
+                if key in params:
+                    if type(params[key]) is not type(default_value):
+                        raise web_request.error("invalid type for %s: %s" % (key, params[key]))
+                    filament_info[key] = params[key]
+
+            self._state[channel] = FILAMENT_DT_STATE_IDLE
+            self._filament_info_update(channel, filament_info, True)
+            web_request.send({'state': 'success'})
+        except Exception as e:
+            logging.error("[filament_detect] set: %s", str(e))
+            web_request.send({'state': 'error', 'message': str(e)})
+
     def get_all_filament_info(self):
         return self._filament_info
 
