--- rootfs.original/home/lava/klipper/klippy/extras/fm175xx_reader.py
+++ rootfs/home/lava/klipper/klippy/extras/fm175xx_reader.py
@@ -244,8 +244,9 @@
         self.__self_test_times = 100
         self.__self_test_success_cnt = 0
 
-        self.__printer.register_event_handler("klippy:ready", self.__ready)
-        self.__printer.register_event_handler("klippy:shutdown", self.__shutdown)
+        if config.getboolean('enabled', True):
+            self.__printer.register_event_handler("klippy:ready", self.__ready)
+            self.__printer.register_event_handler("klippy:shutdown", self.__shutdown)
 
     def __ready(self):
         self.__select_fm175xx_obj(FM175XX_CHANNEL_1)
